app/controllers/assignments_controller.rb

```diff
@@ -110,4 +115,18 @@ def send_create_assignment_statsd_events
     GitHubClassroom.statsd.increment("exercise.create")
     GitHubClassroom.statsd.increment("deadline.create") if @assignment.deadline
   end
+
+  def sort_assignment_repositories(assignment_repos)
+    @current_sort_mode = params[:sort_assignment_repos_by] || DEFAULT_SORT_MODE
+    @all_sort_modes    = SORT_MODES
+
+    case @current_sort_mode
+    when "Assignment acceptance time"
+      assignment_repos.to_a
+    when "Student name"
+      assignment_repos.sort_by { |repo| repo.github_user.name }
```

Comment:
This makes an API call to GitHub for every single repo which is kind of unfortunate. There isn't really a better way to do this right now though.